typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
```

Comment:
Discussion for recommended rule: `'@typescript-eslint/prefer-regexp-exec': 'error'`

Reply:
[@typescript-eslint/prefer-regexp-exec](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-regexp-exec.md)
The performance gain proposed here is negligible in most cases. People prefer to use match with strings, because it's more intuitive.

Reply:
Discussed in person. We are OK with Trevor's proposal to disable this rule, though @rajsite is going to do research into side effects of calling `match`. We can use Trevor's comment as the explanation.

Reply:
Example of how match has side-effects (mutates properties on the RegExp constructor):
![image](https://user-images.githubusercontent.com/1588923/112509566-9a4b4b00-8d5e-11eb-8f9c-04f21bc9b680.png)

Reply:
This thread can be resolved.